newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectWriter.js

```
@@ -0,0 +1,175 @@
+  cloudProjectId,
+}: {
+  authenticatedUser: AuthenticatedUser,
+  project: gdProject,
+  cloudProjectId: string,
+}): Promise<?string> => {
+  const archive = await zipProject(project);
+  const newVersion = await commitVersion(
+    authenticatedUser,
+    cloudProjectId,
+    archive
+  );
+  return newVersion;
+};
+
+export const generateOnSaveProject = (
+  authenticatedUser: AuthenticatedUser
+) => async (project: gdProject, fileMetadata: FileMetadata) => {
+  const newVersion = await zipProjectAndCommitVersion({
+    authenticatedUser,
+    project,
+    cloudProjectId: fileMetadata.fileIdentifier,
+  });
+  if (!newVersion) return { wasSaved: false, fileMetadata };
+  return {
+    wasSaved: true,
+    fileMetadata,
+  };
+};
+
+export const generateOnChangeProjectProperty = (
+  authenticatedUser: AuthenticatedUser
+) => async (
+  project: gdProject,
+  fileMetadata: FileMetadata,
+  properties: { name: string }
+): Promise<boolean> => {
+  if (!authenticatedUser.authenticated) return false;
+  try {
+    await updateCloudProject(
+      authenticatedUser,
+      fileMetadata.fileIdentifier,
+      properties
+    );
+    const newVersion = await zipProjectAndCommitVersion({
+      authenticatedUser,
+      project,
+      cloudProjectId: fileMetadata.fileIdentifier,
+    });
+    if (!newVersion) {
+      throw new Error("Couldn't save project following property update.");
+    }
+
+    return true;
+  } catch (error) {
+    // TODO: Determine if a feedback should be given to user so that they can try again if necessary.
+    console.warn(
+      'An error occurred while changing cloud project name. Ignoring.',
+      error
+    );
+    return false;
+  }
+};
+
+const createCloudProjectAndInitCommit = async (
+  project: gdProject,
+  authenticatedUser: AuthenticatedUser,
+  name: string
+): Promise<?string> => {
+  try {
+    const cloudProject = await createCloudProject(authenticatedUser, {
+      name,
+    });
+    if (!cloudProject) return;
```

Open question: should you throw instead of returning?

---

For clarity's sake, it's better so I changed it. But in the end, it won't appear as an error to the user since it will return `wasSaved: false`